VMBackup/main/freezesnapshotter.py

```diff
@@ -129,8 +129,24 @@ def __init__(self, logger, hutil , freezer, g_fsfreeze_on, para_parser, takeCras
 
             if(para_parser.includedDisks != None and CommonVariables.isAnyDiskExcluded in para_parser.includedDisks.keys()):
                 if (para_parser.includedDisks[CommonVariables.isAnyDiskExcluded] == True and (para_parser.includeLunList == None or para_parser.includeLunList.count == 0)):
-                    self.logger.log('Some disks are excluded from backup and LUN list is not present. Setting the snapshot mode to onlyGuest.')
-                    self.takeSnapshotFrom = CommonVariables.onlyGuest
+                    if( CommonVariables.isAnyDirectDriveDiskIncluded in para_parser.includedDisks.keys() and para_parser.includedDisks[CommonVariables.isAnyDirectDriveDiskIncluded] == True):
+                        errMsg = 'DirectDrive disk is included, so the host must create the snapshot. IsAnyDiskExcluded is true, but, the included LUN list is empty in the extension input, '\
+                                    'which is not allowed for host DoSnapshot. Thus, failing the extension run.'
+                        self.logger.log(errMsg, True, 'Error')
+                        self.hutil.SetExtErrorCode(ExtensionErrorCodeHelper.ExtensionErrorCodeEnum.FailedInputMismatch)
+                    elif( CommonVariables.isVmgsBlobIncluded in para_parser.includedDisks.keys() and para_parser.includedDisks[CommonVariables.isVmgsBlobIncluded] == True):
+                        errMsg = 'VmgsBlob is included, so the host must create the snapshot. IsAnyDiskExcluded is true, but, the included LUN list is empty in the extension input, '\
+                                    'which is not allowed for host DoSnapshot. Thus, failing the extension run.'
+                        self.logger.log(errMsg, True, 'Error')
+                        self.hutil.SetExtErrorCode(ExtensionErrorCodeHelper.ExtensionErrorCodeEnum.FailedInputMismatch)
+                    elif( CommonVariables.isAnyWADiskIncluded in para_parser.includedDisks.keys() and para_parser.includedDisks[CommonVariables.isAnyWADiskIncluded] == True):
+                        errMsg = 'WADisk is included, so the host must create the snapshot. IsAnyDiskExcluded is true, but, the included LUN list is empty in the extension input, '\
+                                    'which is not allowed for host DoSnapshot. Thus, failing the extension run.'
```

Comment:
"IsAnyDiskExcluded is true, but, the included LUN list is empty in the extension input,"
Please describe the reasoning for this in comments. It is not clear.

Reply:
sure will add that